src/services/auth.service.ts

```diff
@@ -8,34 +8,119 @@ import AuthenticationError from "../common/utils/customErrors/authenticationErro
 import { redisClient } from "../storage/redisClient";
 import { authHelper } from "../common/helpers/auth.helper";
 import logger from "../common/logger/logger";
+import { signupDto } from "../data/dtos/signup.dto";
+import { Account } from "../models/account.model";
+import { hashHelper } from "../common/helpers/hash.helper";
+import { cityRepository } from "../repositories/city.repository";
+import clientError from "../common/utils/customErrors/clientError";
 
-class AuthService {
-  login = async (params: loginDto): Promise<string> => {
+export class AuthService {
+  isCityIdValid = async (id: string): Promise<boolean> => {
     try {
-      const account = await accountRepository.getUserAccount(params);
+      logger.info("is CityId Valid", {
+        cityId: id,
+        __filename,
+        functionName: "isCityIdValid",
+      });
+      const getCityId: string | null = await cityRepository.getCityId(id);
+      if (getCityId == null) {
+        throw new clientError("city id doesnot exist", id, 422);
+      } else {
+        return true;
+      }
+    } catch (err) {
+      throw err;
+    }
+  };
 
+  storeToken = async (token: string, userId: string): Promise<void> => {
+    try {
+      logger.info("store token in redis", {
+        userId,
+        __filename,
+        functionName: "storeToken",
+      });
+      await redisClient.setToken(userId, token); // set the JWT as the key and its value as valid
+      const payload = authHelper.verifyAccessToken(token); // verifies and decode the jwt to get the expiration date
+      const exp = payload.exp as number;
+      await redisClient.setExpireAt(userId, exp); // sets the token expiration date to be removed from the cache
```

Comment:
@ShouryaMup-CAW Please use the common function from Redis client. Generalize the input going to set Token with key, value, expiry.